### PR TITLE
Email verification process.  Issue #10

### PR DESCRIPTION
**Pull request by maedae**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
-      request.path != "/users/confirmation" &&
-      request.path != "/companies/create/confirmation" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/authentication" &&
-      request.path != "/companyInvites/verification" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/verification/confirmation" &&
-      !request.xhr?) # don't store ajax calls
-    session[:previous_url] = request.fullpath 
+  def store_location
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Move model logic into model (request use_count > 4)

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
-end
-
-protected
 
-def after_update_path_for(resource)
-  @company = Company.find_by_id(current_user.company_id)
-  session[:company_username] = @company.username
-  user_path(resource)
-end
+  def after_sign_in_path_for(resource)
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Remove unused methods (applicationcontroller#after_sign_in_path_for)

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/authentication" &&
-      request.path != "/companyInvites/verification" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/verification/confirmation" &&
-      !request.xhr?) # don't store ajax calls
-    session[:previous_url] = request.fullpath 
+  def store_location
+    # store last url - this is needed for post-login redirect to whatever the user last visited.
+    return unless request.get?
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Remove trailing whitespace

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
+  def store_location
+    # store last url - this is needed for post-login redirect to whatever the user last visited.
+    return unless request.get? 
+    if (request.path != "/users/sign_in" &&
+        request.path != "/users/sign_up" &&
+        request.path != "/users/password/new" &&
+        request.path != "/users/password/edit" &&
+        request.path != "/users/confirmation" &&
+        request.path != "/companies/create/confirmation" &&
+        request.path != "/companies/create" &&
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Calls request.path != "/companies/create" 2 times (DuplicateMethodCall)

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
-      request.path != "/companyInvites/authentication" &&
-      request.path != "/companyInvites/verification" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/verification/confirmation" &&
-      !request.xhr?) # don't store ajax calls
-    session[:previous_url] = request.fullpath 
+  def store_location
+    # store last url - this is needed for post-login redirect to whatever the user last visited.
+    return unless request.get? 
+    if (request.path != "/users/sign_in" &&
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Calls request.path 11 times (DuplicateMethodCall)

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Has unused parameter 'resource' (UnusedParameters)

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Assignment Branch Condition size for store_location is too high. [43.02/15]

Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Cyclomatic complexity for store_location is too high. [14/6]

Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Method has too many lines. [15/10]

Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Perceived complexity for store_location is too high. [14/7]

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
-      request.path != "/companies/create/confirmation" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/authentication" &&
-      request.path != "/companyInvites/verification" &&
-      request.path != "/companies/create" &&
-      request.path != "/companyInvites/verification/confirmation" &&
-      !request.xhr?) # don't store ajax calls
-    session[:previous_url] = request.fullpath 
+  def store_location
+    # store last url - this is needed for post-login redirect to whatever the user last visited.
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Line is too long. [96/80]

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Trailing whitespace detected.

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Don't use parentheses around the condition of an `if`.

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -5,36 +5,28 @@ class ApplicationController < ActionController::Base
...
+        request.path != "/users/password/edit" &&
+        request.path != "/users/confirmation" &&
+        request.path != "/companies/create/confirmation" &&
+        request.path != "/companies/create" &&
+        request.path != "/companyInvites/authentication" &&
+        request.path != "/companyInvites/verification" &&
+        request.path != "/companies/create" &&
+        request.path != "/companyInvites/verification/confirmation" &&
+        !request.xhr?) # don't store ajax calls
+      session[:previous_url] = request.fullpath
```
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Trailing whitespace detected.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (sumeetjain) [inline app/controllers/application_controller.rb]: Unused method argument - `resource`. If it's necessary, use `_` or `_resource` as an argument name to indicate that it won't be used. You can also write as `after_sign_in_path_for(*)` if you want the method to accept any arguments but don't care about them.

File: app/mailers/user_mailer.rb [hunk 8]
```diff
@@ -1,4 +1,5 @@
 class UserMailer < ApplicationMailer
+  default :from => 'noreply@fyiio.com'
```
Comment (sumeetjain) [inline app/mailers/user_mailer.rb:2]: Use the new Ruby 1.9 hash syntax.

File: config/environments/development.rb [hunk 9]
```diff
@@ -1,20 +1,20 @@
...
-  config.cache_classes = false
-  config.eager_load = false
-  config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
-  config.action_mailer.raise_delivery_errors = true
-  config.after_initialize do
-    Bullet.enable = true
-    Bullet.bullet_logger = true
-    Bullet.rails_logger = true
+  Rails.application.configure do
```
Comment (sumeetjain) [inline config/environments/development.rb:1]: Indentation of first line in file detected.

**Comment by maedae** [comment]
Don't need this branch any longer.

**Pull request merged**